### PR TITLE
Refactor widget base

### PR DESCRIPTION
**Pull request by bzang**
A big clean up of the base widget along with some refactoring of other related modules to be actual HOCs vs just helper functions. The base widget is now in `spark-widget-base`. The recents widget has been migrated to this, but the other widgets need some extra work to get to there due to a bug in the original `widget-base` package.

Still needed
- [x] Data API
- [x] Update Readme

### CODE REVIEW

**Comment by adamweeks** [comment]
Deploy preview ready!

Built with commit aba9869bb98ce32d1c923b9a6fccc9c94e0fd9b5

https://deploy-preview-594--hopeful-hamilton-4090b9.netlify.com

File: packages/node_modules/@ciscospark/spark-widget-base/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+> **NOTE**: Be sure to attach a `displayName` to your widget. This will be used to add your widget to the global namespace.
+
+### Widgets using Redux
+
+If you are also using `redux` for state management and want to provide your own reducers, import and use the `constructSparkEnhancer` higher order component instead:
+
+``` js
+import React, {Component} from 'react';
+import {constructSparkWidget} from '@ciscospark/spark-widget-base';
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/spark-widget-base/README.md]: should be `import {constructSparkEnhancer}`

**Comment by adamweeks** [comment]
Looks good so far. Would love to see a bit more in the readme for widget base on the available options.

**Comment by bzang** [comment]
Good call. I wrote the README before I started the dev, and some things definitely changed just because of technical limitations.

**Comment by bzang** [comment]
@adamweeks, ready for another round. I'm going to spin a different PR to get the existing widgets migrated over.

**Pull request merged**